src/base/static/components/molecules/tag-editor.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+import { translate } from "react-i18next";
+
+import { TextareaInput } from "../atoms/input";
+import TagName from "./tag-name";
+
+import mapseedApiClient from "../../client/mapseed-api-client";
+
+const TagContainer = styled("div")(props => ({
+  outline: "none",
+  boxSizing: "border-box",
+  width: "100%",
+  padding: "8px",
+  display: "flex",
+  alignItems: "center",
+  marginBottom: "8px",
+  borderTop: props.isSelected ? "1px solid #a8a8a8" : "1px dashed #aaa",
+  borderRight: props.isSelected ? "1px solid #c8c8c8" : "1px dashed #aaa",
+  borderBottom: props.isSelected ? "1px solid transparent" : "1px dashed #aaa",
+  borderLeft: props.isSelected ? "1px solid #c8c8c8" : "1px dashed #aaa",
+  borderRadius: "3px",
+  backgroundColor: props.isSelected
+    ? props.backgroundColor || "#6495ed"
+    : "transparent",
+
+  "&:hover": {
+    cursor: "pointer",
+    backgroundColor: props.backgroundColor || "#6495ed",
+    opacity: props.isSelected ? 1 : 0.3,
+  },
+}));
+
+const NoteBox = styled(TextareaInput)(props => ({
+  outline: "none !important",
+  padding: "0 8px 0 6px",
+  overflow: props.isFocused ? "visible" : "hidden",
+  height: props.isFocused ? "5rem" : "0.7rem",
+  lineHeight: props.isFocused ? "1rem" : "0.8rem",
+  background: "transparent",
+  borderLeft: "1px solid #fff !important",
+  borderRight: "none !important",
+  borderTop: "none !important",
+  borderBottom: "none !important",
+  color: "#fff",
+  fontStyle: "italic",
+  fontSize: "0.75rem",
+  fontWeight: "normal",
+  transition: "all .5s ease",
```

Comment:
I thought we decided to put our CSS inside the atom, and make it configurable through an API instead of overriding the raw CSS. This will make theming more manageable when we re-use the TextareaInput atom.

Reply:
sure will do

Reply:
Though I'm not sure we want all these properties in the atom, as some of them are pretty specific to this particular use of a textarea (like the transition and overflow settings). I think I'll keep some of them here and make the rest configurable at the atom level if that sounds ok?

Reply:
Sounds good. Ideally we have an api that takes a prop like "animate" that set's all of those, instead of having the caller configure all the CSS and override the class name.